### PR TITLE
feat!: replace checkValidity with query + assertions

### PR DESCRIPTION
**Pull request by tjwelde**
## fixes KILTProtocol/ticket#2106
This PR removes the `Attestation.checkValidity` function.
Instead developers have to query themselves and just check for null and if it is revoked or not.

## How to test:
- Tests

## Checklist:

- [x] I have verified that the code works
- [x] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [x] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: packages/core/src/__integrationtests__/Attestation.spec.ts [hunk 1]
```diff
@@ -173,7 +173,9 @@ describe('When there is an attester, claimer and ctype drivers license', () => {
       tokenHolder.address
     )
     await submitExtrinsic(authorizedStoreTx, tokenHolder)
-    expect(await Attestation.checkValidity(attestation.claimHash)).toBe(true)
+    const storedAttestation = await Attestation.query(attestation.claimHash)
+    expect(storedAttestation).not.toBeNull()
```
Comment (arty-name) [inline packages/core/src/__integrationtests__/Attestation.spec.ts:177]: Do you think we need this double check? Would the next line fail when undefined !== false?

**Review by arty-name — APPROVED** [review]

File: packages/core/src/__integrationtests__/Attestation.spec.ts (hunk 1 above)
Comment (rflechtner) [reply packages/core/src/__integrationtests__/Attestation.spec.ts:177]: agreed, seems like `(await Attestation.query(attestation.claimHash)?.revoked === false)` could be our recommended check to see whether an attestation is valid

**Review by rflechtner — APPROVED** [review]
LGTM

File: packages/core/src/__integrationtests__/Attestation.spec.ts (hunk 1 above)
Comment (tjwelde) [reply packages/core/src/__integrationtests__/Attestation.spec.ts:177]: yeah, was thinking the same. Just added it for completeness.

**Pull request merged**